tests/NumberTest.php

```diff
@@ -0,0 +1,497 @@
+            ['0.1', '', '0', '1', '0.1'],
+            ['1.0', '', '1', '0', '1'],
+            ['1.1', '', '1', '1', '1.1'],
+            ['1.234', '', '1', '234', '1.234'],
+            ['0.1245', '', '0', '1245', '0.1245'],
+            ['1', '', '1', '0', '1'],
+            ['01', '', '1', '0', '1'],
+            ['01.0', '', '1', '0', '1'],
+            ['01.01', '', '1', '01', '1.01'],
+            ['10.2345', '', '10', '2345', '10.2345'],
+            [
+                '123917549171231.12451028401824',
+                '',
+                '123917549171231',
+                '12451028401824',
+                '123917549171231.12451028401824'
+            ],
+            ['+12351.49273592', '', '12351', '49273592', '12351.49273592'],
+            ['-12351.49273592', '-', '12351', '49273592', '-12351.49273592'],
+            ['-12351', '-', '12351', '0', '-12351'],
+            ['-0', '', '0', '0', '0'],
+            ['-01', '-', '1', '0', '-1'],
+            ['-01.0', '-', '1', '0', '-1'],
+            ['-01.01', '-', '1', '01', '-1.01'],
+        ];
+    }
+
+    public function provideValidExponents()
+    {
+        return [
+            'exponent 0' => ['123456', 0, '123456'],
+            'exponent 1' => ['123456', 1, '12345.6'],
+            'exponent 2' => ['123456', 2, '1234.56'],
+            'exponent 3' => ['123456', 3, '123.456'],
+            'exponent 4' => ['123456', 4, '12.3456'],
+            'exponent 5' => ['123456', 5, '1.23456'],
+            'exponent 6' => ['123456', 6, '0.123456'],
+            'exponent 7' => ['123456', 7, '0.0123456'],
+            'exponent 8' => ['123456', 8, '0.00123456'],
+            'exponent 8 with leading' => ['0123456', 8, '0.00123456'],
+            'zero' => ['0', 8, '0'],
+            'leading zeroes' => ['00000', 8, '0'],
+            'plus leading zeroes' => ['+00000', 8, '0'],
+            'minus exponent 0' => ['-123456', 0, '-123456'],
+            'minus exponent 1' => ['-123456', 1, '-12345.6'],
+            'minus exponent 2' => ['-123456', 2, '-1234.56'],
+            'minus exponent 3' => ['-123456', 3, '-123.456'],
+            'minus exponent 4' => ['-123456', 4, '-12.3456'],
+            'minus exponent 5' => ['-123456', 5, '-1.23456'],
+            'minus exponent 6' => ['-123456', 6, '-0.123456'],
+            'minus exponent 7' => ['-123456', 7, '-0.0123456'],
+            'minus exponent 8' => ['-123456', 8, '-0.00123456'],
+            'minus exponent 8 with leading' => ['-0123456', 8, '-0.00123456'],
+            'minus zero' => ['-0', 8, '0'],
+            'minus leading zeroes' => ['-00000', 8, '0'],
+        ];
+    }
+
+    public function provideInvalidNumbers()
+    {
+        return [
+            'bool false' => [false],
+            'bool true' => [true],
+            'empty string' => [''],
+            'NaN' => ['asd'],
+            'NaN with dot' => ['asd.foo'],
+            'NaN with point' => ['asd.foo'],
+            'array' => [array()],
```

Comment:
This time I did it on purpose to make it more evident that I'm using the array as a parameter